pyrdp/player/ReplayThread.py

```diff
@@ -107,4 +107,11 @@ def setSpeed(self, speed: float):
         self.queue.put(ReplayThreadEvent.SPEED)
 
     def close(self):
-        self.queue.put(ReplayThreadEvent.EXIT)
\ No newline at end of file
+        self.queue.put(ReplayThreadEvent.EXIT)
+
+    def parentResized(self):
+        """
+        Seeks to current time to allow to rerender the image on the resized widget.
+        """
+        self.lastSeekTime = self.timer.getElapsedTime()
```

Comment:
Um, I don't think seeking should be necessary to support scaling.

This is likely an indicator that the logic is not being handled in the right place. We have a `QRemoteDesktop` widget that always has the latest rendered frame in a `QImage` buffer. The easiest way to handle the scaling would be to propagate the resize event all the way to it, and have it keep the original size buffer and paint the scaled buffer whenever a window repaint is requested.

Does what I'm saying make sense? I have google a little bit and found that people are successfully propagating the resize event down their widget tree, so this doesn't sound impossible.

In the worst case scenario, I don't mind manual propagation all the way to the QRemoteDesktop widget, but I think that that's where it should be handled.

Reply:
Okay, I've read a bit more. It seems like events will bubble up, but not drill down into the tree, so your code handling resize by calling a resize handler on the children is unavoidable AFAICT. But the rest of this comment still stands.

Reply:
This is already the case. QRemoteDesktop's `_buffer` contains the unscaled image, which is then rescaled at render time.

This is a good catch. Your comment made me realise that I only did this because the previous implementation created a new blank image upon changing the scaling, but I can simply copy the previous buffer onto the new one, which removes the need to seek.